installer/Templates/Head.php

```diff
@@ -0,0 +1,103 @@
+        <link rel="icon" type="image/png" href="installer/assets/img/favicon/favicon-32x32.png" sizes="32x32"/>
+        <link rel="icon" type="image/png" href="installer/assets/img/favicon/favicon-96x96.png" sizes="96x96"/>
+        <link href="installer/assets/css/style.css" rel="stylesheet"/>
+        
+        <!-- @yield(\'style\') -->
+    </head>
+    <body>
+        <div class="master">
+            <div class="box">
+                <div class="header">
+                    <h1 class="header__title">'.$input['title'].'</h1>
+                </div>
+                <ul class="step">
+                    <li class="step__divider"></li>
+                    <li class="step__item '.($input['step'] == 'Migrate' ? 'active' : '')
+			.'"  title="Creating the Database">';
+		echo '<i class="step__icon fa fa-server" aria-hidden="true"></i>';
+		echo '
+                    </li>';
+		echo '
+                    <li class="step__divider"></li>
+                    <li class="step__item '.($input['step'] == 'Env' ? 'active' : '')
+			.'"  title="Setting the environment">';
+		if ($this->get_index($input['step']) >= 3) {
+			echo '<a href="?step=env">
+                                <i class="step__icon fa fa-cog" aria-hidden="true"></i>
+                            </a>';
+		} else {
+			echo '<i class="step__icon fa fa-cog" aria-hidden="true"></i>';
+		}
+		echo '</li>
+                    <li class="step__divider"></li>
+                    <li class="step__item '.($input['step'] == 'Permissions' ? 'active'
+				: '').'"  title="Checking Permissions">';
+
+		if ($this->get_index($input['step']) >= 2) {
+			echo '<a href="?step=perm"><i class="step__icon fa fa-key" aria-hidden="true"></i></a>';
+		} else {
+			echo '<i class="step__icon fa fa-key" aria-hidden="true"></i>';
+		};
+		echo '
+                    </li>
+                    <li class="step__divider"></li>
+                    <li class="step__item '.($input['step'] == 'Requirements' ? 'active'
+				: '').'" title="Checking Requirements">';
+
+		if ($this->get_index($input['step']) == 1) {
+			echo '<a href="?step=req"><i class="step__icon fa fa-list" aria-hidden="true"></i></a>';
+		} else {
+			echo '<i class="step__icon fa fa-list" aria-hidden="true"></i>';
+		};
+
+		echo '</li>
+                    <li class="step__divider"></li>
+                    <li class="step__item '.($input['step'] == 'Welcome' ? 'active' : '')
+			.'"  title="Welcome!">';
+			echo '<a href="?step="><i class="step__icon fa fa-home" aria-hidden="true"></i></a>';
+
+		echo '
+                    </li>
+                    <li class="step__divider"></li>
+                </ul>
+                <div class="main">';
+
+		if (isset($input['errors']))
+		{
+			echo '
+		            <p class="alert alert-danger text-center">
+		                <strong>Please fix the errors before going to the next step.</strong>
```

Comment:
I think that in case of errors being detected, there should be button the user can click to rerun the test that failed (you know, the user fixes the permission or whatnot in the background and then wants to continue). Right now I can click on the buttons at the top, but they don't actually rerun the tests; they just display the same thing. I need to reload the whole page in the browser, which is not exactly user-friendly.

Reply:
will see what I can do.